crates/sui-types/src/transaction.rs

```diff
@@ -863,17 +863,17 @@ impl Command {
     }
 }
 
-fn write_sep<T: Display>(
+pub(crate) fn write_sep<T: Display>(
     f: &mut Formatter<'_>,
     items: impl IntoIterator<Item = T>,
     sep: &str,
 ) -> std::fmt::Result {
     let mut xs = items.into_iter().peekable();
     while let Some(x) = xs.next() {
+        write!(f, "{x}")?;
         if xs.peek().is_some() {
             write!(f, "{sep}")?;
         }
-        write!(f, "{x}")?;
     }
     Ok(())
 }
```

Comment:
Not really an issue introduced by this PR, but a better way to write this would be:

```rust
let mut xs = items.into_iter();
let Some(x) = xs.next() else {
    return Ok(());
}
write!(f, "{x}");
for x in xs {
    write!(f, "{sep}{x}")?;
}
Ok(())
```

No requirement on peeking, no test at every iteration of the loop.

Reply:
I added this improvement in #15957